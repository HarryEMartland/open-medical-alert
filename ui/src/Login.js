import React from 'react';
import ReCAPTCHA from "react-google-recaptcha";


function Login({onLogin, loading, setPassword, password, errorMessage, captchaChange, valid}) {

    return <div className='container-fluid'>
        <div className='row justify-content-md-center pt-5 text-center'>
            <div className='col-sm col-md-6 col-lg-4 col-xl-3'>
                <form onSubmit={onLogin} >
                    {errorMessage && <div className="alert alert-danger" role="alert">{errorMessage}</div>}
                    <div className="form-group">
                        <input type='password' className='form-control form-control-lg' disabled={loading} onChange={setPassword}
                               value={password} placeholder='Password'/>
                    </div>
                    <div className="form-group">
                        <ReCAPTCHA
                            sitekey={process.env.REACT_APP_CAPTCHA_SITE_KEY}
                            onChange={captchaChange}
                            onExpired={()=>{captchaChange()}}
                        />
                    </div>
                    <div className="form-group">
                        <input type='submit' className='btn btn-success btn-lg' disabled={!valid} value='Login'/>
                    </div>
                </form>
            </div>
        </div>
    </div>
}

export default Login