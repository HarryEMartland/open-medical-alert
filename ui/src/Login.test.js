import React from "react";
import {render, unmountComponentAtNode} from "react-dom";
import {act} from "react-dom/test-utils";
import Login from "./Login";

describe('Login', () => {

    let container = null;
    beforeEach(() => {
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('should render with data', () => {

        act(() => {
            render(<Login/>, container);
        })
        expect(container).toMatchSnapshot();

    })

    it('should fire callback when submit is clicked', () => {

        const onLoginCallback = jest.fn();

        act(() => {
            render(<Login onLogin={onLoginCallback} valid={true}/>, container);
        });

        container.querySelector("[type='submit']").dispatchEvent(new MouseEvent("click", {bubbles: true}));

        expect(onLoginCallback).toHaveBeenCalled();
    })

    it('should disable when loading', () => {

        act(() => {
            render(<Login loading={true}/>, container);
        })
        expect(container).toMatchSnapshot();
    })

    it('should render error message', ()=>{
        act(() => {
            render(<Login errorMessaGE='THIS IS AN ERROR MESSAGE'/>, container);
        })
        expect(container).toMatchSnapshot();
    })
});