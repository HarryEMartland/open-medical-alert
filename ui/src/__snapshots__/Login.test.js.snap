// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should disable when loading 1`] = `
<div>
  <div
    class="container-fluid"
  >
    <div
      class="row justify-content-md-center pt-5 text-center"
    >
      <div
        class="col-sm col-md-6 col-lg-4 col-xl-3"
      >
        <form>
          <div
            class="form-group"
          >
            <input
              class="form-control form-control-lg"
              disabled=""
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <div />
          </div>
          <div
            class="form-group"
          >
            <input
              class="btn btn-success btn-lg"
              disabled=""
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Login should render error message 1`] = `
<div>
  <div
    class="container-fluid"
  >
    <div
      class="row justify-content-md-center pt-5 text-center"
    >
      <div
        class="col-sm col-md-6 col-lg-4 col-xl-3"
      >
        <form>
          <div
            class="form-group"
          >
            <input
              class="form-control form-control-lg"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <div />
          </div>
          <div
            class="form-group"
          >
            <input
              class="btn btn-success btn-lg"
              disabled=""
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Login should render with data 1`] = `
<div>
  <div
    class="container-fluid"
  >
    <div
      class="row justify-content-md-center pt-5 text-center"
    >
      <div
        class="col-sm col-md-6 col-lg-4 col-xl-3"
      >
        <form>
          <div
            class="form-group"
          >
            <input
              class="form-control form-control-lg"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <div />
          </div>
          <div
            class="form-group"
          >
            <input
              class="btn btn-success btn-lg"
              disabled=""
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
